aws-sdk-go/leo.go

```diff
@@ -0,0 +1,252 @@
+        Key     : aws.String(FileName),
+    })
+    if getObjErr != nil {
+        log.Printf("Get Object %s/%s Failed, %s\n", BucketName, FileName, getObjErr)
+        return
+    }
+    content, _ := ioutil.ReadAll(getObjRes.Body)
+    if !bytes.Equal(content, basicContent) {
+        log.Printf("Get Object %s/%s Failed, Content NOT Match\n", BucketName, FileName)
+        return
+    }
+    log.Printf("Get Object %s/%s Suceeded\n", BucketName, FileName)
+    log.Println()
+
+    largeContentPart := largeContent[1048576:10485760+1]
+
+    rngSingleRes, rngSingleErr := svc.GetObject(&s3.GetObjectInput{
+        Bucket  : bucketStr,
+        Key     : aws.String(singleFileName),
+        Range   : aws.String("bytes=1048576-10485760"),
+    })
+    if rngSingleErr != nil {
+        log.Printf("Range Get Single Part Object %s/%s Failed, %s\n", BucketName, singleFileName, rngSingleErr)
+        return
+    }
+    rngSingleCon, _ := ioutil.ReadAll(rngSingleRes.Body)
+    if !bytes.Equal(rngSingleCon, largeContentPart){
+        log.Printf("Range Get Single Part Object %s/%s Failed, Content NOT Match\n", BucketName, singleFileName)
+        return
+    }
+    log.Printf("Range Get Single Part Object %s/%s Suceeded\n", BucketName, singleFileName)
+    log.Println()
+
+    rngMpRes, rngMpErr := svc.GetObject(&s3.GetObjectInput{
+        Bucket  : bucketStr,
+        Key     : aws.String(mpFileName),
+        Range   : aws.String("bytes=1048576-10485760"),
+    })
+    if rngMpErr != nil {
+        log.Printf("Range Get Multi Part Object %s/%s Failed, %s\n", BucketName, mpFileName, rngMpErr)
+        return
+    }
+    rngMpCon, _ := ioutil.ReadAll(rngMpRes.Body)
+    if !bytes.Equal(rngMpCon, largeContentPart){
+        log.Printf("Range Get Multi Part Object %s/%s Failed, Content NOT Match\n", BucketName, mpFileName)
+        return
+    }
+    log.Printf("Range Get Multi Part Object %s/%s Suceeded\n", BucketName, mpFileName)
+    log.Println()
+
+    listObjRes, listObjErr := svc.ListObjects(&s3.ListObjectsInput{Bucket: bucketStr})
+    if listObjErr != nil {
+        log.Printf("List Object %s/ Failed, %s\n", BucketName, listObjErr)
+        return
+    }
+    log.Printf("List Object Res:\n")
+    for _, obj := range listObjRes.Contents {
+        log.Printf("Key: %s\t, Size: %d\t, ETag: %s\n", *obj.Key, *obj.Size, *obj.ETag)
+    }
+
+    _, delObjErr := svc.DeleteObject(&s3.DeleteObjectInput{
+        Bucket  : bucketStr,
+        Key     : aws.String(FileName),
+    })
+    if delObjErr != nil {
+        log.Printf("Delete Object %s/%s Failed, %s\n", BucketName, FileName, delObjErr)
+        return
+    }
+    log.Printf("Delete Object %s/%s Suceeded\n", BucketName, FileName)
+    log.Println()
+
+    _, err = svc.DeleteBucket(&s3.DeleteBucketInput{Bucket: bucketStr})
+    if err != nil {
+        log.Printf("Delete Bucket %s Failed, %s\n", BucketName, err)
+        return
+    }
+    log.Printf("Delete Bucket %s Suceeded\n", BucketName)
+}
```

Comment:
As you might notice,
The current aws-client-tests suites written in ONE function due to some reason...
I think we need to refactor code with the manner one function per one test case 
for making it easy to maintain.
Please consider that.